analyzers/tests/SonarAnalyzer.UnitTest/TestCases/AvoidExcessiveClassCoupling.CSharp10.cs

```diff
@@ -9,7 +9,8 @@ class Foo1 : FooBase { }
     public struct MyStruct { }
 
     public record struct FirstRecordStruct { }
-    public record struct SecondRecordStruct // FN
+    public record struct SecondRecordStruct // Noncompliant {{Split this record into smaller and more specialized ones to reduce its dependencies on other types from 6 to the maximum authorized 1 or less.}}
```

Comment:
Should this read *Split this record **struct** into smaller...* instead?

Reply:
It would make sense. Can you do it in a very short piece of code? I'm bit afraid it would require too much logic for a minor improvement.

Reply:
The name is retrieved via `typeDeclaration.Keyword.ValueText` which lives in `Microsoft.CodeAnalysis`. It could be done, but it is probably isn't worth it.

https://github.com/SonarSource/sonar-dotnet/blob/99412a1d7e61f87286242edfff17f1c58e354a24/analyzers/src/SonarAnalyzer.CSharp/Rules/AvoidExcessiveClassCoupling.cs#L108-L109